wgpu-hal/src/vulkan/device.rs

```diff
@@ -1954,11 +1954,11 @@ impl crate::Device for super::Device {
         match module {
             super::ShaderModule::Raw(raw) => {
                 unsafe { self.shared.raw.destroy_shader_module(raw, None) };
+
+                self.counters.shader_modules.sub(1);
```

Comment:
I think this is the wrong fix to the problem, we should always be incrementing the counter, no matter what kind of shader module.